components/icon-button/index.js

```diff
@@ -25,14 +25,21 @@ class IconButton extends Component {
 		const classes = classnames( 'components-icon-button', className );
 		const tooltipText = tooltip || label;
 
-		// Should show the tooltip if an explicit tooltip is passed
-		// or if there's a label and the children are empty and the tooltip is not explicitely disabled
-		const showTooltip = !! tooltip ||
+		// Should show the tooltip...
+		const showTooltip = (
+			// if an explicit tooltip is passed or...
```

Comment:
I find this style of commenting each conditional really useful, especially after working on a patch around https://github.com/WordPress/gutenberg/blob/master/editor/components/block-list/block.js#L425 👍🏻

Reply:
A nitpick is moving the "if" above might make more sense, eg:

```js
// Should show the tooltip if...
  const showTooltip = (
    //an explicit tooltip is passed or...
```

Reply:
So move all "if"s? "Or if..."?

Reply:
I think moving the if to the top and ending each line with `or...` or `and...`, along with the indentation, would make sense.

But what you have already makes sense, I just think the wording could be tidied up a bit. If my suggestion is less clear than I could be wrong and should be ignored 😅

Reply:
I'd say it could be:

```js
// Should show the tooltip if...
const showTooltip = (
	// an explicit tooltip is passed or...
	tooltip ||
	// there's a shortcut or...
	shortcut ||
	(
		// there's a label and...
		label &&
		// the children are empty and...
		( ! children || ( isArray( children ) && ! children.length ) ) &&
		// the tooltip is not explicitely disabled.
		false !== tooltip
	)
);
```

(Includes the lack of `!!` prefixes.)